### DISCUSSION TITLE
Using __base__ and __config__ in create_model

### DISCUSSION THREAD

**Discussion by s1beria21** [pydantic/pydantic#6826] [Question] [answered]
i have few classes
```python
class MyBase(BaseModel):
    name: str

class MyClass(MyBase):
    id: int

    class Config:
        from_attributes = True
```

I want dynamicly create this classes
```python
MyBase = create_model("MyBase", name=(str, ...))
MyClass = create_model("MyClass", id=(int, ...), __base__=MyBase, __config__={'from_attributes': True})
```
but i recieve error:
```
pydantic.errors.PydanticUserError: to avoid confusion `__config__` and `__base__` cannot be used together
```

Is there way avoid this error?

**Comment by Kludex** [comment]
No... It exists, so you don't do it... 🤔

**Reply by zmievsa** [reply]
Hi, @Kludex ! Could you help me understand **why** we don't want to have this behavior in pydantic? We have that in regular model creation so why not have it in dynamic model creation too?

**Comment by braindevices** [comment]
well, it does not allowed

**Comment by Typiqally** [comment]
One thing I did instead is to add `model_config` as a field defintion. In your case it would look something like this:

```python
MyBase = create_model("MyBase", name=(str, ...))
MyClass = create_model("MyClass", id=(int, ...), model_config=(ConfigDict(from_attributes=True)), __base__=MyBase)
```

```python
print(MyClass.model_config)
# returns {'from_attributes': True}
```

**Comment by martinherrerias** [comment]
I had the same issue with pydantic 2.10.6, upgrading to 2.11.5 seems to have fixed it.

**Comment by Viicos** [comment, marked as answer]
Fixed in https://github.com/pydantic/pydantic/pull/11714, introduced in 2.11.4.
